### DISCUSSION TITLE
Installation on Dardel

### DISCUSSION THREAD

**Discussion by timofeymukha** [ExtremeFLOW/neko#539] [Show and tell]
* Make sure you use Cray, `ml PrgEnv-cray`
* Clone as usual and `cd neko`
* `cp /usr/share/aclocal/pkg.m4 m4/`
* `ml PDC`
* `ml buildtools`
* `./regen.sh`
* Select prefix for the install path you want `./configure --prefix=path_to_install_neko_install CC=cc MPIFC=ftn`
* `make && make install`

**Comment by timofeymukha** [comment]
To avoid issues with checkpoints, add `ml cce/11.0.4` after `ml PrgEnv-cray`

**Reply by njansson** [reply]
Latest develop or v0.4.1, should (I hope) address this for `cce/13.0.0`

**Reply by njansson** [reply]
The `pkg.m4` file is now included in the repo, so the third step could be avoided

**Comment by vbaconnet** [comment]
Most recent compilation that works for me :) 

```bash

ml craype-accel-amd-gfx90a rocm/6.2.4 PrgEnv-cray
module use /cfs/klemming/pdc/projects/hpcrd/modules
ml hpcrd
ml json-fortran/8.3.0-cce-18.0.1-bjoug3p


./regen.sh
./configure MPIFC=ftn MPICC=cc MPICXX=CC FCFLAGS="-m4 -O2" --prefix=$MY_PREFIX HIPCC=hipcc FC=ftn CC=cc --with-hip=/opt/rocm-6.0.3 HIP_HIPCC_FLAGS="-O3 --offload-arch=gfx90a" --enable-device-mpi
 make -j && make install
 ```

**Reply by vbaconnet** [reply]
`-m4` suppresses the warnings from the Cray compiler

**Reply by shiyu-du-dsy** [reply]
rocm/6.3.3 works for me

**Reply by shiyu-du-dsy** [reply]
Is` --with-hip=/opt/rocm-6.0.3` a good practice if we put it in the tutorial? I mean I usually try barely with `--with-hip`, but first I am not sure about the default setting, and second I am not sure whether we should keep the name here to be the same as our module loading of rocm :)

**Reply by shiyu-du-dsy** [reply]
Another stuff I would like to bring about: if one would like to use api on gpu, then rocm/6.1.3 should be loaded due to the requirement of both `.so.5` file and `.so.6` file

**Comment by vbaconnet** [comment]
```
ml craype-accel-amd-gfx90a rocm PrgEnv-cray
module use /cfs/klemming/pdc/projects/hpcrd/modules
ml hpcrd 
ml json-fortran/9.3.1-cce20

./regen.sh
./configure FC=ftn CC=cc MPIFC=ftn MPICC=cc MPICXX=CC CFLAGS="-O3" FCFLAGS="-O3 -m4 -hlist=am" --prefix=installdir --with-hip=${ROCM_PATH} HIPCC=hipcc HIP_HIPCC_FLAGS="-O3 -ffast-math --offload-arch=gfx90a" --enable-device-mpi
```
